compaction/compaction.cc

```diff
@@ -137,14 +139,28 @@ std::string_view to_string(compaction_type_options::scrub::quarantine_mode quara
 
 static api::timestamp_type get_max_purgeable_timestamp(const table_state& table_s, sstable_set::incremental_selector& selector,
         const std::unordered_set<shared_sstable>& compacting_set, const dht::decorated_key& dk, uint64_t& bloom_filter_checks,
-        const api::timestamp_type compacting_max_timestamp) {
+        const api::timestamp_type compacting_max_timestamp, is_shadowable is_shadowable) {
     if (!table_s.tombstone_gc_enabled()) [[unlikely]] {
         return api::min_timestamp;
     }
 
     auto timestamp = api::max_timestamp;
-    auto memtable_min_timestamp = table_s.min_memtable_timestamp();
-    // Use memtable timestamp if it contains data older than the sstables being compacted,
+    api::timestamp_type memtable_min_timestamp;
+    if (is_shadowable) {
+        // For shadowable tombstones, check the minimum live row_marker timestamp
+        // as it shadows any older shadowable tombstones, exposing live cells with
```

Comment:
s/it shadows/such a row would shadow/ (row marker timestamp doesn't shadow anything

Reply:
okay